lib/html_renderer.rb

```diff
@@ -1,39 +1,79 @@
 module Gobstones
   class HtmlRenderer
+    def initialize(options)
+      @options = options
+    end
+
     def render_success(result)
-      @result = {
-        boards: prepare_boards([:initial, :final], result)
-      }
+      bind_result(
+        boards: prepare_boards([:initial, :final], result),
+        reason: prepare_reason(result[:reason])
+      )
+    end
 
-      bind
+    def render_error_check_final_board_failed_different_boards(result)
+      bind_result(
+        error: :check_final_board_failed_different_boards,
+        boards: prepare_boards([:initial, :expected, :actual], result)
+      )
     end
 
-    def render_error_check_final_board_failed(result)
-      @result = {
-        error: :check_final_board_failed,
-        boards: prepare_boards([:actual, :initial, :expected], result)
-      }
+    def render_error_check_final_board_failed_unexpected_boom(result)
+      bind_result(
+        error: :check_final_board_failed_unexpected_boom,
+        boards: prepare_boards([:initial, :expected, :actual], result),
+        reason: prepare_reason(result[:reason])
+      )
+    end
+
+    def render_error_check_error_failed_expected_boom(result)
+      bind_result(
+        error: :check_error_failed_expected_boom,
+        boards: prepare_boards([:initial, :expected, :final], result)
+      )
+    end
 
-      bind
+    def render_error_check_error_failed_another_reason(result)
+      bind_result(
+        error: :check_error_failed_another_reason,
+        error_parameter: I18n.t("code_#{result[:expected_code]}"),
+        reason: prepare_reason(result[:reason])
+      )
     end
 
     private
 
+    def prepare_reason(reason)
+      return if not reason
+      reason[:message]
+    end
+
     def prepare_boards(names, result)
-      names.map { |it|
-        struct title: "#{it}_board".to_sym, board: adapt_to_view(result[it])
+      names.reject { |it|
+        it == :initial and not @options[:show_initial_board]
+      }.map { |it|
+        struct title: "#{it}_board".to_sym,
+               board: (
+                  if result[it] == :boom.to_s
```

Comment:
Is the `result[it]` always a string? In that case, wither write `result[it].to_sym == :boom` or `result[it] == 'boom'`